packages/workflow/src/nexus.ts

```diff
@@ -0,0 +1,208 @@
+  /**
+   * The operation's service name.
+   */
+  readonly service: string;
+
+  /**
+   * The name of the Operation.
+   */
+  readonly operation: string;
+
+  /**
+   * Operation token as set by the Operation's handler. May be empty if the operation completed synchronously.
+   */
+  readonly token?: string;
+
+  /**
+   * Wait for Operation completion and get its result.
+   */
+  result(): Promise<T>;
+}
+
+/**
+ * Options for {@link createNexusClient}.
+ */
+export interface NexusClientOptions<T> {
+  endpoint: string;
+  service: T;
+}
+
+/**
+ * Create a Nexus client for invoking Nexus Operations from a Workflow.
+ *
+ * @experimental Nexus support in Temporal SDK is experimental.
+ */
+export function createNexusClient<T extends nexus.ServiceDefinition>(options: NexusClientOptions<T>): NexusClient<T> {
+  class NexusClientImpl<T extends nexus.ServiceDefinition> implements NexusClient<T> {
+    async executeOperation<O extends T['operations'][keyof T['operations']]>(
+      operation: string | T['operations'][nexus.OperationKey<T['operations']>],
+      input: nexus.OperationInput<T['operations'][nexus.OperationKey<T['operations']>]>,
+      operationOptions?: Partial<StartNexusOperationOptions>
+    ): Promise<nexus.OperationOutput<O>> {
+      const handle = await this.startOperation(operation, input, operationOptions);
+      return await handle.result();
+    }
+
+    async startOperation<O extends T['operations'][keyof T['operations']]>(
+      operation: string | T['operations'][nexus.OperationKey<T['operations']>],
+      input: nexus.OperationInput<T['operations'][nexus.OperationKey<T['operations']>]>,
+      operationOptions?: StartNexusOperationOptions
+    ) {
+      const opName = typeof operation === 'string' ? options.service.operations[operation].name : operation.name;
+
+      const activator = getActivator();
+      const seq = activator.nextSeqs.nexusOperation++;
+
+      const execute = composeInterceptors(
+        activator.interceptors.outbound,
+        'startNexusOperation',
+        startNexusOperationNextHandler
+      );
+
+      // TODO: Do we want to make the interceptor async like we do for child workflow? That seems redundant.
+      // REVIEW: I ended up changing this so that the interceptor returns a Promise<StartNexusOperationOutput>,
+      //         and the result promise is contained in that Output object. As a consequence of this,
+      //         the result promise/completion does not exist until the StartNexusOperation event is received.
+      //         That's totally different from what we did in ChildWorkflow, but I think that's cleaner from
+      //         interceptors point of view, and will make it easier to extend the API in the future.
+      const { token, result: resultPromise } = await execute({
+        endpoint: options.endpoint,
+        service: options.service.name,
+        operation: opName,
+        options: operationOptions ?? {},
+        nexusHeader: {},
```

Comment:
Do you want to make it possible for the user to pass headers now, or leave an open issue for it?

Reply:
Also I know Go uses the (weird) singular; does TS also? Python doesn't.

Reply:
> Also I know Go uses the (weird) singular for "header"; does TS also? Python [uses](https://github.com/temporalio/sdk-python/blob/main/temporalio/workflow.py#L5212-L5213) "headers".

Good catch. Renamed to `headers` (no `nexus` prefix, plural form). That's coherent with Python and other TS interceptors.

Reply:
> Do you want to make it possible for the user to pass headers now, or leave an open issue for it?

Not sure what's the expectation here. For all other APIs, headers are not exposed directly in the "main" user facing call, only in interceptors. Are doing this differently for Nexus ops? If that's only in interceptors, then that should already be working now (though no test atm covering that).

In any case, I'm leaving that for later.